rio_tiler_pds/sentinel/aws/sentinel2.py

```diff
@@ -262,13 +265,19 @@ def __attrs_post_init__(self):
         self.bounds = self.stac_item["bbox"]
         self.crs = WGS84_CRS
 
-        self.bands = tuple(
-            [
-                sentinel_l2a_band_map.get(band)
-                for band in self.stac_item["assets"]
-                if sentinel_l2a_band_map.get(band)
-            ]
-        )
+        if self.stac_item["stac_version"] == "1.0.0-beta.2":
+            self.bands = tuple(
+                [band for band in default_l2a_bands if band in self.stac_item["assets"]]
+            )
+
+        else:
+            self.bands = tuple(
+                [
+                    band
+                    for band, name in sentinel_l2a_band_map.items()
+                    if name in self.stac_item["assets"]
+                ]
+            )
```

Comment:
I've decided to go the other way around! We assume only the bands in `default_l2a_bands` are possible for Sentinel-2 L2A so we just validate that they exists in the assets (either using the band name) or using the new asset name if the Stac Item is not `1.0.0-beta.2` (Maybe we could open the test to `1.0.0-beta.1` too 🤷

Reply:
Makes sense, thanks for picking this up! 🙏